packages/css/src/plugins/__tests__/sort-at-rules.test.ts

```diff
@@ -0,0 +1,845 @@
+                  color: abc;
+                }
+                @media screen and (width < 500px) and (width > 1000px) {
+                  color: abc;
+                }
+                @media screen and (width < 500px) and (width > 1000px) or (height > 1000px) {
+                  color: abc;
+                }
+              "
+      `);
+    });
+
+    it('should sort @container queries just like media queries', () => {
+      const actual = transform(`
+        @media (min-width: 400px) {
+          color: purple;
+        }
+        @media (min-width: 200px) {
+          color: blue;
+        }
+        @container summary (height > 200px) and (width > 400px) {
+          color: yellow;
+        }
+        @container tall (height > 300px) {
+          color: green;
+        }
+        @container sidebar (height > 200px) {
+          color: orange;
+        }
+        @container (width > 400px) {
+          color: red;
+        }
+      `);
+
+      expect(actual).toMatchInlineSnapshot(`
+        "
+                @container (width > 400px) {
+                  color: red;
+                }
+                @container sidebar (height > 200px) {
+                  color: orange;
+                }
+                @container summary (height > 200px) and (width > 400px) {
+                  color: yellow;
+                }
+                @container tall (height > 300px) {
+                  color: green;
+                }
+                @media (min-width: 200px) {
+                  color: blue;
+                }
+                @media (min-width: 400px) {
+                  color: purple;
+                }
+              "
+      `);
+    });
+
+    it('should sort non-media queries too', () => {
+      // We make no assumption here on whether other parts of Compiled actually support
+      // all of these... but we might as well sort them consistently.
+      const actual = transform(`
+        @supports (display: flex) {
+          color: blue;
+        }
+        @media (300px <= height <= 400px) {
+          color: yellow;
+        }
+        @property --item-size {
+          syntax: "<percentage>";
+          inherits: true;
+          initial-value: 20%;
+        }
+        @layer module, state;
+        @media (300px < height <= 400px) {
+          color: indigo;
+        }
+        @layer state {
+          p {
+            color: orange;
+          }
+        }
+        @media (min-height: 400px) {
+          color: purple;
+        }
+        @supports (color: red) {
+          color: red;
+        }
+        @media (300px <= height) {
+          color: green;
+        }
+        @layer module {
+          p {
+            color: pink;
+          }
+        }
+      `);
+
+      expect(actual).toMatchInlineSnapshot(`
+        "
+                @layer module {
+                  p {
+                    color: pink;
+                  }
+                }
+                @layer module, state;
```

Comment:
What's the rule for sorting layers?

Reply:
Oh, interesting… it's worth noting that sorting `@layer` would be antithetical to the w3 spec as the definition-based order is heavily important…but does this ever fit into atomic css, or should layers require some `@keyframes` type hoisting where they must be globally ordered by another means, possibly manually by the user.

Does not impact us, just might be worth a note in a changeset or some page that atomic css and `@layer` conflict.

…that being said, again, going from possibly non-deterministic to deterministic, and the solution before and after this sorting would always be to define `@layer` upfront—if anyone's actually using Compiled with layers, they must do this, or they probably don't know what they're doing.  LGTM.